### PR TITLE
[Merged by Bors] - feat: add lemmas about closed and quotient maps

### PR DESCRIPTION
**Pull request by urkud**
Forward-port of leanprover-community/mathlib#19071

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Maps.lean [hunk 1]
```diff
@@ -278,6 +278,12 @@ theorem quotientMap_iff {α β : Type _} [TopologicalSpace α] [TopologicalSpace
   and_congr Iff.rfl topologicalSpace_eq_iff
 #align quotient_map_iff quotientMap_iff
 
+theorem quotientMap_iff_closed [TopologicalSpace α] [TopologicalSpace β] {f : α → β} :
```
Comment (eric-wieser) [inline Mathlib/Topology/Maps.lean:281]: Can you remove the `{α β : Type _}` from the lemma above too, as in the diff?

File: Mathlib/Topology/Separation.lean [hunk 2]
```diff
@@ -1330,6 +1330,12 @@ theorem Continuous.closedEmbedding [CompactSpace α] [T2Space β] {f : α → β
   closedEmbedding_of_continuous_injective_closed h hf h.isClosedMap
 #align continuous.closed_embedding Continuous.closedEmbedding
 
+/-- A surjective continuous map from a compact space to a Hausdorff space is a quotient map. -/
```
Comment (eric-wieser) [inline Mathlib/Topology/Separation.lean:1333]: nit: the lemma above says "A continuous injective map" in mathlib4 and "An injective continuous map" in mathlib3. Can you make the two lemmas match, either by using the mathlib3 or mathlib4 spelling in both places?

**Review by eric-wieser — APPROVED** [review]
bors d+

Two minor nits

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**